src/pages/LandingPage.tsx

```diff
@@ -29,65 +29,68 @@ import {usePaginatedGallery} from "../components/Card/usePaginatedGallery";
 
 import { GridContent } from './GridContent';
 import { ListContent } from './ListContent';
-import {usePagination} from "../hooks/usePagination";
+import { useLandingConfigStore, usePaginationStore, defaultAvailablePerPage } from '../hooks/useStore';
 import {Config} from "../config";
 
 export const LandingPage: FunctionComponent = () => {
-  const [searchInput, setSearchInput] = useState('');
-  const [view, setView] = useState<'grid'|'list'>('grid');
+  const searchInput = useLandingConfigStore((state) => state.searchInput)
+  const setSearchInput = useLandingConfigStore((state) => state.updateInput)
+
+  const selectedTags = useLandingConfigStore((state) => state.selectedTags)
+  const setSelectedTags = useLandingConfigStore((state) => state.updateSelectedTags)
+
+  const view = useLandingConfigStore((state) => state.view)
+  const setView = useLandingConfigStore((state) => state.updateView)
 
   const onChange = (searchInput: string) => {
     setSearchInput(searchInput);
   };
 
-  const [selectedTags, setSelectedTags] = useState<ReadonlyArray<string>>([]);
-
   const filteredDocs = useMemo(() => apiConfigurations
       .filter((apiConfig) => apiConfig.displayName.toLowerCase().includes(searchInput.toLowerCase()))
       .filter(apiConfig => selectedTags.length === 0 || apiConfig.tags.some(tag => selectedTags.includes(tag.id))),
   [searchInput, selectedTags]
   );
 
   const galleryId = 'apid-c-api-gallery';
-  const pagination = usePagination(filteredDocs, 10);
+
+  const pagination = usePaginationStore();
 
   usePaginatedGallery(galleryId, view === 'grid', {
-    setPage: pagination.onSetPage,
+    setPage: pagination.setPage,
+    page: pagination.page,
     perPage: pagination.perPage,
-    setPerPage: pagination.onSetPerPage,
+    setPerPage: pagination.setPerPage,
     setAvailablePerPage: pagination.setAvailablePerPage,
-    defaultAvailablePerPage: usePagination.defaultAvailablePerPage
+    defaultAvailablePerPage: defaultAvailablePerPage,
+    elements: filteredDocs,
+    setItems: pagination.setItems,
   });
 
   const changeView = (toView: 'grid' | 'list') => {
     setView(toView);
-    pagination.onSetPage(1);
+    pagination.setPage(1);
     if (toView === 'list') {
-      pagination.setAvailablePerPage();
-      pagination.onSetPerPage(10);
+      pagination.setAvailablePerPage(defaultAvailablePerPage);
+      pagination.setPerPage(10);
     }
   }
 
   const clearFilters = () => {
     setSearchInput('');
     setSelectedTags([]);
-    pagination.onSetPage(1);
+    pagination.setPage(1);
   };
 
-  useEffect(() => {
-    const onSetPage = pagination.onSetPage;
-    onSetPage(1);
-  }, [filteredDocs, pagination.onSetPage]);
```

Comment:
I think we need this one or similar - it's supposed to change the page to 1 if the elements change. Is it resolved elsewhere?

Reply:
We seem to be updating the `filteredDocs` at least once every time the LandingPage component loads regardless of whether or not there was an update to the elements array. Not sure where exactly we are doing that, probably in the `useMemo()` section.
I changed this approach to a more decentralized one, so every time there is a change in the tags and in the search input, we are setting the page to 1. I think those are the only times we will need to set the page to 1.
Lmk, if you think there is a better approach here :+1: